aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -314,20 +329,24 @@ public Cancelable observe(
             },
             onObservationFailure,
             onObservationCompleted
-        );
+        )));
     }
 
-    @SuppressWarnings("checkstyle:WhitespaceAround") // () -> {}
-    @NonNull
     @Override
-    public <T extends Model> Cancelable observe(
+    public <T extends Model> void observe(
             @NonNull Class<T> itemClass,
             @NonNull QueryPredicate selectionCriteria,
+            @NonNull Consumer<Cancelable> onObservationStarted,
             @NonNull Consumer<DataStoreItemChange<T>> onDataStoreItemChange,
             @NonNull Consumer<DataStoreException> onObservationFailure,
             @NonNull Action onObservationCompleted) {
         onObservationFailure.accept(new DataStoreException("Not implemented yet, buster!", "Check back later!"));
-        return () -> {};
+    }
+
+    private void afterInitialization(@NonNull final Runnable runnable) {
```

Comment:
I like this refactor out of the individual overrides, basically lifecycle checks. I wonder do we need a separate `beforeInitialization` to ensure everything is in a good state? Could that be used for the race condition issue we saw the other day on app startup instead of requiring a singleton?

Reply:
Hm, yea we could. Right now, `afterInitialization` is just only a method in this implementation class, not anything fancier. And there logically _is_ some `beforeInitialization`, stuff (basically, setup a listener). Right now, the `configure(...)` is being used for both configuration and for this "before" initialization step.